### ISSUE TITLE
Revive doesn't analyze tests in separate _test packages

### ISSUE THREAD

**Issue by alexandear** [mgechev/revive#1357]
**Describe the bug**

`revive` ignores tests located in `_test` packages and doesn't analyze them.

**To Reproduce**

1. Consider the following project with any tests in a `_test` package:

<details><summary>Details</summary>
<p>

```
❯ tree .
.
├── bugtestpkg_test.go
├── bugtestpkg.go
└── go.mod

1 directory, 3 files
```

```
-- go.mod --
module bugtestpkg

go 1.23

-- bugtestpkg.go --
// Package bugtestpkg is a package.
package bugtestpkg

// Add returns a + b.
func Add(a, b int) int {
        return a + b
}

-- bugtestpkg_test.go --
package bugtestpkg_test

import (
        "bugtestpkg"
        "testing"
)

func Id() {}

func TestAdd(t *testing.T) {
        if bugtestpkg.Add(1, 2) != 3 {
                t.FailNow()
        }
}
```

</p>
</details> 

2. Run the latest `revive` without any configuration:

```shell
revive ./...
```

**Expected behavior**

There should be one warning:

```
bugtestpkg_test.go:7:6: func Id should be ID
```

**Actual behavior**

No warnings are reported.

**Desktop**

<details><summary>Details</summary>
<p>

```
❯ go env
AR='ar'
CC='cc'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='c++'
GCCGO='gccgo'
GO111MODULE=''
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/Users/alexandear/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/alexandear/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/q0/5_z6pvw574z1c0zcrr2xvk0r0000gn/T/go-build1859025823=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/Users/alexandear/src/github.com/mgechev/revive/bugtestpkg/go.mod'
GOMODCACHE='/Users/alexandear/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/alexandear/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/opt/homebrew/Cellar/go/1.24.3/libexec'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/alexandear/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/opt/homebrew/Cellar/go/1.24.3/libexec/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.24.3'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</p>
</details> 

**Additional context**

First noticed in #1353.

Golangci-lint correctly detects the issue:

```
❯ golangci-lint run --enable-only revive
bugtestpkg/bugtestpkg_test.go:8:6: var-naming: func Id should be ID (revive)
func Id() {}
     ^
1 issues:
* revive: 1
```

**Comment by chavacava** [comment]
The problem seems to be a bug in mgechev/dots.
Resolving files from a package relies on `go/build` but `dots` does not retrieve all the files yield by `go/build`. More specifically at

https://github.com/mgechev/dots/blob/e955255bf51709c96049fd17a712991643566bc1/resolve.go#L115-L119

```go
	var files []string
	files = append(files, pkg.GoFiles...)
	files = append(files, pkg.CgoFiles...)
	files = append(files, pkg.TestGoFiles...)
	if pkg.Dir != "." {
```

`dots` doesn't add files under `pkg.XTestGoFiles` to teh final list of files.
That code section shoud be 

https://github.com/chavacava/dots/blob/584c67cb2f80750d350ce61f641d43d22f561d96/resolve.go#L115-L120

```go
	var files []string
	files = append(files, pkg.GoFiles...)
	files = append(files, pkg.CgoFiles...)
	files = append(files, pkg.TestGoFiles...)
	files = append(files, pkg.XTestGoFiles...)
	if pkg.Dir != "." {
```


I have a branch of `dots` with the fix and building `revive` with it fixes the bug described in this PR
